test/unit/host_discovered_test.rb

```diff
@@ -324,15 +324,15 @@ class HostDiscoveredTest < ActiveSupport::TestCase
   test "all non-discovery facts are deleted after managed conversion" do
     Setting[:discovery_clean_facts] = true
     raw = parse_json_fixture('regular_host', true)
-    raw.merge!({
-      'delete_me' => "content",
-      'discovery_keep_me' => "content",
-      })
+    raw.merge!({ 'discovery_keep_me' => "content" })
     host = discover_host_from_facts(raw)
+    fact_name = FactoryBot.create(:fact_name, name: 'PuppetFact', type: 'PuppetFactName')
+    FactoryBot.create(:fact_value, fact_name: fact_name, value: 'delete_me', host: host)
     host.save
     managed = ::ForemanDiscovery::HostConverter.to_managed(host)
+    assert_equal host.facts_hash['PuppetFact'], 'delete_me'
     managed.clear_facts
-    assert_nil managed.facts_hash['delete_me']
+    assert_nil host.facts_hash['PuppetFact']
     assert_equal "content", managed.facts_hash['discovery_keep_me']
   end
```

Comment:
Can you revert this test completely to the original state? If I am not mistaken, no change should be required in this case so it should be passing. This way we will be sure we are not changing any behavior.